### ISSUE TITLE
Details of out-of-distribution

### ISSUE THREAD

**Issue by cheny02** [TreeLLi/APT#7]
Thank you for sharing your work and code!

In the paper, you claim to use M= 4 and 16 shots when testing the generalization of prompts.

But in the weights you give, it is M= 1 and 100 shots, could you please clarify the detailed experimental setting of out-of-distribution (such as M, shots, batch, and epoch)?

Thanks

**Comment by TreeLLi** [comment]
Hi,

thank you for your interest in our work. For OOD experiments, M=4 and 16 shots were used. The weights you mentioned, M=1 and 100 shots, were used for zero-shot experiments. We have not released the weights of models used for OOD experiments. If you need those, we could try to find them but not sure if they are still available since the file system of our server recently crashed.

Best wishes,
Lin

**Comment by cheny02** [comment]
Hi, Thank you for your response.

What is the difference between zero-shot experiments and OOD experiments since both of them trained on Imagenet and tested on downstream tasks.

I would appreciate it if you could provide the weights for the OOD experiments. If those are not available, the batch and epoch would also be helpful. I can reproduce it myself.

Thanks！

**Comment by TreeLLi** [comment]
Hi,

they differ in this work regarding the data distribution gap between training and test data. OOD involves the distribution shift only in the input space so the training and test datasets have the consistent category system, while zero-shot involves the distribution shift in both input space and output category so the test input data and/or category may never be seen in the training data. 

The detailed experimental setups are described in the appendix. Please let me know if there are something missing in the appendix.

Lin

**Comment by cheny02** [comment]
So the cross datasets experiments is the zero-shot experiments right?

Thanks for your reply!

**Comment by TreeLLi** [comment]
yes, we follow the zero-shot setup in this work: https://arxiv.org/abs/2212.07016. Kindly please close the issue if you feel it is resolved, otherwise, I am happy to answer any other question you have.

**Comment by cheny02** [comment]
Hi, sorry to bother again.

As mentioned above, the OOD experiments employ M=4 with 16-shot learning, while the zero-shot configuration uses M=1 with 100-shot. However, upon examining the provided model weights (specifically zs_imagenet_vitb32_eps4_nctx1_100shots_UC), I noticed that this checkpoint contains OOD evaluation results that closely align with the reported values in the manuscript. So I am wondering if the published performance corresponds to the 16-shot or 100-shot setting.

Thanks

**Comment by TreeLLi** [comment]
Hi,

thanks for reaching out. The published results correspond to the specified setups in the paper.

**Comment by cheny02** [comment]
Thanks
